src/MongoDB.Driver.Core/Core/Configuration/ConnectionString.cs

```diff
@@ -457,11 +457,11 @@ public ConnectionString Resolve()
             var host = GetHostNameForDns();
 
             var client = new LookupClient();
-            var response = client.Query(srvPrefix + host, QueryType.SRV);
+            var response = client.Query(srvPrefix + host, QueryType.SRV, QueryClass.IN);
```

Comment:
So the type of `response` has changed from 1.0.7 to 1.2.0?

Reply:
Query still returns an instance of IDnsQueryResponse.

But IDnsQueryResponse has an Answers property that used to be of type ICollection<DnsResourceRecord> and is now of type IReadOnlyList<DnsResourceRecord>.

That's the change that there is no way of working around.